pkg/nfd-master/nfd-master.go

```diff
@@ -605,19 +605,74 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// validateExtendedResources validates extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, resourceLabels map[string]string) map[string]string {
+	extendedResources := make(map[string]string)
+	for extendedResource, capacity := range resourceLabels {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, m.deniedNs.wildcard, m.deniedNs.normal) {
+			if _, ok := m.args.ExtraLabelNs[ns]; !ok {
+				klog.Errorf("Namespace %q is not allowed. Ignoring Extended Resource  %q\n", ns, extendedResource)
+				continue
```

Comment:
I think we could have a static deny list for now. Anyway keep it separate from the labels (the config option is also named `denyLabelNs`).